lib/auth/join/join.go

```diff
@@ -260,6 +260,10 @@ type RegisterResult struct {
 // running on a different host than the auth server. This method requires a
 // provision token that will be used to authenticate as an identity that should
 // be allowed to join the cluster.
+//
+// Deprecated: this function is superceded by lib/join/joinclient.Join
+//
+// TODO(nklaassen): DELETE IN 19.0.0
```

Comment:
im not sure how many versions we go ahead for auth, in clients we usually have to keep something for at least n+1 versions for backward compatibility. is this ok to remove in the next major?

Reply:
switched this to 20, see https://github.com/gravitational/teleport/pull/59778#discussion_r2399586844